src/main/java/com/elastic/support/diagnostics/commands/SystemCallsCmd.java

```diff
@@ -1,25 +1,232 @@
+            return;
+        } else if (context.isDocker()) {
+            // If docker is running and we got an id passed in check to see if it's
+            // in the list of container id's we obtained in the host check. If it is
+            // just get that one. If it's not there or there was no id go get all the containers in the list.
+            List<String> containers = context.getDockerContainers();
+            String dockerId = context.getDiagnosticInputs().getDockerId();
+            if (StringUtils.isNotEmpty(dockerId)) {
+                boolean isDockerIdInList = containers.contains(dockerId);
+                if (isDockerIdInList) {
+                    containers.clear();
+                    containers.add(dockerId);
+                }
+            }
+
+            // Run the non-container specific docker calls
+            Map<String, String> dockerGlobal = context.getDiagsConfig().getCommandMap("docker-global");
+            processCalls(tempDir, dockerGlobal, pb);
+
+            Map<String, String> dockerContainers = context.getDiagsConfig().getCommandMap("docker-containers");
+            // Now get the stats for each container in the list. If they passed one in and we found it this will
+            // be a single iteration
+            for (String container : containers) {
+                try {
+                    String dir = context.getTempDir() + SystemProperties.fileSeparator + container;
+                    Files.createDirectories(Paths.get(dir));
+                    Map<String, String> containerCalls = preProcessDockerCommands(dockerContainers, container);
+                    processCalls(dir, containerCalls, pb);
+                } catch (IOException e) {
+                    logger.log(SystemProperties.DIAG, e);
+                    logger.info("Error creating container directory - bypassing Docker container calls.");
+                }
+            }
+        } else {
+            String calls = checkOS();
+            Map<String, String> osCmds = context.getDiagsConfig().getCommandMap(calls);
+            osCmds = preProcessOsCmds(osCmds, pid, SystemProperties.javaHome);
+            processCalls(tempDir, osCmds, pb);
+        }
+
+    }
+
+    public void processCalls(String targetDir, Map<String, String> commandMap, ProcessBuilder pb) {
+
+        commandMap.forEach((k, v) -> {
+                    runCommand(targetDir, k, v, pb);
+                }
+        );
+    }
+
+    public String checkOS() {
+        String osName = SystemProperties.osName.toLowerCase();
+        if (osName.contains("windows")) {
+            return "winOS";
+        } else if (osName.contains("linux")) {
+            return "linuxOS";
+        } else if (osName.contains("darwin") || osName.contains("mac os x")) {
+            return "macOS";
+        } else {
+            logger.error("Failed to detect operating system!");
+            throw new RuntimeException("Unsupported OS");
```

Comment:
I wonder if we should fail or try to assume `linuxOS`?

Reply:
It's been like this forever and I don't think I've ever seen an instance where this actually happened. And if it was some kind of freaky self build OS flavor that didn't yield something expected I'm not sure we should make the assumption. But on reflection it does seem appropriate to run the rest. In the next PR I'll have it logging the specific value encountered and bypassing system calls.

Reply:
That's fair. I was thinking of Solaris FWIW.